> FWIW, refactoring to use a MethodHandle is overkill. The cost was in looking up the `Method` every time, not the actual invocation. Saving the returned `Method` from `findMethod` would be enough.

Thanks for the suggestion, however after testing (logging the exceptions) I've discovered it hasn't worked since 1.17 due to JPMS restrictions and ModLauncher limitations, so I've removed it altogether:

`Caused by: java.lang.IllegalAccessError: class net.minecraftforge.registries.ForgeRegistry tried to access private method 'void java.util.BitSet.trimToSize()' (net.minecraftforge.registries.ForgeRegistry is in module forge@47.1.50-1-20-optimise-forgeregistry-validatecontent of loader 'TRANSFORMER' @29b40b3; java.util.BitSet is in module java.base of loader 'bootstrap')`

--- a/src/main/java/net/minecraftforge/registries/ForgeRegistry.java
+++ b/src/main/java/net/minecraftforge/registries/ForgeRegistry.java
@@ -5,9 +5,6 @@
 
 package net.minecraftforge.registries;
 
-import java.lang.invoke.MethodHandle;
-import java.lang.invoke.MethodHandles;
-import java.lang.invoke.MethodType;
 import java.util.*;
 import java.util.Map.Entry;
 import java.util.function.IntFunction;

